### PR TITLE
Minor/id code cleanup

### PR DESCRIPTION
**Pull request by derobins**
Cleans up H5I code:

- Consistent and improved naming
- Splits code into multiple files like other packages
- Quiets const warnings
- Cleans up includes

### CODE REVIEW

File: src/H5Itest.c [hunk 1]
```diff
@@ -11,10 +11,8 @@
  * help@hdfgroup.org.                                                        *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
-/* Programmer:  Quincey Koziol
- *              Tuesday, July 27, 2010
- *
- * Purpose:	ID testing functions.
+/* 
+ * H5Iint.c - ID testing functions
```
Comment (qkoziol) [inline src/H5Itest.c]: H5Itest.c

**Review by qkoziol — APPROVED** [review]
There's some package-scoped variables that would be nice to resolve with package-scoped routines, but I can totally understand the decisions here and they are very reasonable.

File: src/H5Itest.c (hunk 1 above)
Comment (derobins) [reply src/H5Itest.c]: Oops. Will fix before merge.

**Comment by derobins** [comment]
> There's some package-scoped variables that would be nice to resolve with package-scoped routines, but I can totally understand the decisions here and they are very reasonable.

Yeah, I thought about wrapping those up but figured I'd save that work for later. It would certainly get done in the thread-safety work, for example.

**Review by lrknox — APPROVED** [review]

**Pull request merged**